From 4199626f53ec4ec6f354b88e9435115cd5cf0de9 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Thu, 28 Aug 2014 17:34:02 +0800
Subject: [PATCH 0916/1074] ENGR00329198-02 Revert "ENGR00318938-4 serial:
 imx: add support for no_console_suspend"

The patch "ENGR00318938-4" is a workaround for no_console_suspend support
in M/F mix feature kernel, which is not reasonable that driver enable
uart as wakeup source in voluntary.

This reverts commit c8a2b65c18b1fb280cb0b2a0e65e8fcb9db3bef3.

Signed-off-by: Fugang Duan <B38611@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/tty/serial/imx.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index efee649..919e2d7 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1860,9 +1860,6 @@ static int serial_imx_suspend(struct platform_device *dev, pm_message_t state)
 	struct imx_port *sport = platform_get_drvdata(dev);
 	unsigned int val;
 
-	if (!console_suspend_enabled)
-		enable_irq_wake(sport->port.irq);
-
 	/* enable wakeup from i.MX UART */
 	val = readl(sport->port.membase + UCR3);
 	val |= UCR3_AWAKEN;
@@ -1908,9 +1905,6 @@ static int serial_imx_resume(struct platform_device *dev)
 	val &= ~UCR3_AWAKEN;
 	writel(val, sport->port.membase + UCR3);
 
-	if (!console_suspend_enabled)
-		disable_irq_wake(sport->port.irq);
-
 	uart_resume_port(&imx_reg, &sport->port);
 
 	return 0;
-- 
1.7.5.4

